apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/PooledSession.cs

```diff
@@ -55,14 +55,22 @@ public sealed class PooledSession : IDisposable
         public ByteString TransactionId { get; }
 
         /// <summary>
-        /// The mode of the transaction. (Always None iff TransactionId is null.)
+        /// The options for the transaction that is or will be associated with this session. Won't be null.
         /// </summary>
-        internal ModeOneofCase TransactionMode { get; }
+        /// <remarks>
+        /// Will be <see cref="TransactionOptions.ModeOneofCase.None"/> if <see cref="TransactionId"/>
+        /// is null.
+        /// </remarks>
+        internal TransactionOptions TransactionOptions { get; }
 
         /// <summary>
-        /// The read timestamp of the transaction. (Always <c>null</c> if
-        /// ReturnReadTimestamp = false or if TransactionMode != ReadOnly.)
+        /// The read timestamp of the transaction. May be null.
         /// </summary>
+        /// <remarks>
+        /// Will be set iif <see cref="TransactionOptions.ReadOnly"/> have <see cref="Types.ReadOnly.ReturnReadTimestamp"/>
```

Comment:
have => has? And possibly rewrite it to

```csharp
> /// iif <see cref="TransactionOptions>" represents read-only options with <see cref="Types.ReadOnly.ReturnReadTimestamp"/>
> set to true, of if [...]
```

(And change "and a values was provide" to "and a value was provided".)

I'm curious about the ReturnReadTimestamp bit though... if the options say what kind of transaction the session *will* represent, but we haven't already begun the transaction, how do we have a timestamp?

What about if it's read-only with a specified timestamp in the timestamp bound?